### PR TITLE
Feat/#28 Create SfCheckbox

### PR DESCRIPTION
**Pull request by leomp12**
# Related issue

https://github.com/DivanteLtd/storefront-ui/issues/28

# Scope of work

Edit `SfCheckbox` atom component proposed from PR #59 
- Component API changed:
  - Slots changed;
  - Removed props used to set element classes;
  - Renamed some props;
  - New `disabled` prop;
- SCSS fixes;
- Md reference and stories for `SfCheckbox`;

# Screenshots of visual changes

# Checklist

- [x] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: src/components/atoms/SfCheckbox/SfCheckbox.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        <!-- @slot Custom checked icon image markup (binds `checked` boolean) -->
+        <slot name="icon" :checked="isChecked">
+          <img
+            class="sf-checkbox__icon-svg"
+            src="/assets/check.svg"
+            alt="Check icon" />
+        </slot>
+      </span>
+      <!-- @slot Custom checkbox label text (binds `checked` boolean) -->
+      <slot :checked="isChecked" />
```
Comment (filrak) [inline src/components/atoms/SfCheckbox/SfCheckbox.html]: I guess this slot should have a name

Comment (leomp12) [reply src/components/atoms/SfCheckbox/SfCheckbox.html]: Do you have a name suggestion? This slot may be used in almost all cases, and I think the others will almost never be used, so I left it as the default.

File: src/components/atoms/SfCheckbox/SfCheckbox.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<div class="sf-checkbox">
+  <input
+    :id="id"
+    :name="name"
+    type="checkbox"
+    class="sf-checkbox__field"
+    :value="true"
+    v-model="isChecked"
+    :disabled="disabled" />
+  <!-- @slot Custom checkbox label markup (binds `checked` boolean) -->
```
Comment (filrak) [inline src/components/atoms/SfCheckbox/SfCheckbox.html]: shouldn this be just "custom checkbox markup"? it's not only a label so the slot name is not accurate too. Probably this should be the default slot

File: src/components/atoms/SfCheckbox/SfCheckbox.html (hunk 1 above)
Comment (filrak) [reply src/components/atoms/SfCheckbox/SfCheckbox.html]: Maybe label?

**Comment by filrak** [comment]
I guesss we should ahve similar api to this one https://github.com/DivanteLtd/storefront-ui/pull/91

File: src/components/atoms/SfCheckbox/SfCheckbox.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { shallowMount } from "@vue/test-utils";
+import SfCheckbox from "./SfCheckbox.vue";
+
+describe("SfCheckbox.vue", () => {
+  it("renders a component", () => {
+    const component = shallowMount(SfCheckbox);
+    expect(component.contains(".sf-component")).toBe(true);
```
Comment (filrak) [inline src/components/atoms/SfCheckbox/SfCheckbox.spec.ts]:
```suggestion
    expect(component.contains(".sf-checkbox")).toBe(true);
```

Comment (leomp12) [reply src/components/atoms/SfCheckbox/SfCheckbox.spec.ts]: had not edited the tests yet, now it's ready

**Pull request merged**